token-tracker/src/truth-sources/token-staking.js

```diff
@@ -0,0 +1,187 @@
+/** @typedef { import("../lib/context.js").Context } Context */
+/** @typedef { import("../lib/ethereum-helper").Address } Address */
+
+import { ITruthSource, AddressesResolver } from "./truth-source.js"
+import { Contract } from "../lib/contract-helper.js"
+import { logger } from "../lib/winston.js"
+import { getPastEvents } from "../lib/ethereum-helper.js"
+import { dumpDataToFile } from "../lib/file-helper.js"
+
+import { EthereumHelpers } from "@keep-network/tbtc.js"
+const { callWithRetry } = EthereumHelpers
+
+import Web3 from "web3"
+const { toBN } = Web3.utils
+
+import TokenStakingJSON from "@keep-network/keep-core/artifacts/TokenStaking.json"
+
+const TOKEN_STAKING_HISTORIC_STAKERS_DUMP_PATH =
+  "./tmp/token-staking-stakers.json"
+const TOKEN_STAKING_BALANCES_DUMP_PATH = "./tmp/token-staking-balances.json"
+const TOKEN_STAKING_UNKNOWN_OWNERS_CONTRACTS_DUMP_PATH =
+  "./tmp/token-staking-unknown_owners_contracts.json"
+
+// TODO: Add support for the old TokenStaking contract.
```

Comment:
Correct. The logic should be identical, so we can probably just parameterize this code and call it twice.

Reply:
069c0cf7eaf8c63dd779a3ee41c50d9fdd7311b1